### PR TITLE
Fixes and unit test updates to OpenAI client

### PR DESCRIPTION
**Pull request by efajardo-nv**
## Description
- Update unit tests to handle recently added `api_key` and `base_url` parameters.
- Ensure log directory exists for logging FileHandler.
- Docstring update

## By Submitting this PR I confirm:
- I am familiar with the [Contributing Guidelines](https://github.com/nv-morpheus/Morpheus/blob/main/docs/source/developer_guide/contributing.md).
- When the PR is ready for review, new or existing tests cover these changes.
- When the PR is ready for review, the documentation is up to date with these changes.

### CODE REVIEW

File: tests/llm/services/test_openai_chat_client.py [hunk 1]
```diff
@@ -24,13 +24,30 @@
...
 @pytest.mark.parametrize("max_retries", [5, 10])
-def test_constructor(mock_chat_completion: tuple[mock.MagicMock, mock.MagicMock], max_retries: int):
+def test_constructor(mock_chat_completion: tuple[mock.MagicMock, mock.MagicMock],
+                     api_key: str,
+                     base_url: str,
+                     max_retries: int):
+    client = OpenAIChatClient(OpenAIChatService(api_key=api_key, base_url=base_url),
+                              model_name="test_model",
+                              max_retries=max_retries)
+    assert isinstance(client, LLMClient)
```
Comment (mdemoret-nv) [inline tests/llm/services/test_openai_chat_client.py:37]: This is unnecessary.

**Review by mdemoret-nv — CHANGES_REQUESTED** [review]

File: morpheus/llm/services/openai_chat_service.py [hunk 2]
```diff
@@ -326,8 +328,9 @@ def __init__(self, *, api_key: str = None, base_url: str = None, default_model_k
             The API key for the LLM service, by default None. If `None` the API key will be read from the
             `OPENAI_API_KEY` environment variable. If neither are present an error will be raised.
         base_url : str, optional
-            The api host url, by default None. If `None` the url will be read from the `OPENAI_API_BASE` environment
-            variable. If neither are present the OpenAI default will be used., by default None
+            The api host url, by default None. If the `OPENAI_BASE_URL` environment variable is present,
+            it will always take precedence over this parameter. If neither are present the OpenAI default will
+            be used., by default None
```
[the remark this answers is not in the document]
Comment (efajardo-nv) [reply morpheus/llm/services/openai_chat_service.py:333]: That actually is the current pattern. I had a bad pydantic model config that wasn't allowing the `base_url` to be set. I'll fix the docstring.

File: tests/llm/services/test_openai_chat_client.py [hunk 3]
```diff
@@ -14,23 +14,79 @@
...
+        arg_api_key = "arg-12345"
+        client = OpenAIChatService(api_key=arg_api_key).get_client(model_name="test_model")
+        assert client._client.api_key == arg_api_key
+        assert str(client._client.base_url) == "https://api.openai.com/v1/"
+
+        # Test when both api_key and base_url are passed
+        arg_base_url = "http://arg.openai.com/v1/"
+        client = OpenAIChatService(api_key=arg_api_key, base_url=arg_base_url).get_client(model_name="test_model")
+        assert client._client.api_key == arg_api_key
+        assert str(client._client.base_url) == arg_base_url
```
Comment (mdemoret-nv) [inline tests/llm/services/test_openai_chat_client.py:89]: Lets remove this section and revert back to what it was.

File: tests/llm/services/conftest.py [hunk 4]
```diff
@@ -36,7 +36,7 @@ def openai_fixture(openai):
     yield openai
 
 
-@pytest.fixture(name="mock_chat_completion", autouse=True)
+@pytest.fixture(name="mock_chat_completion", autouse=False)
```
Comment (mdemoret-nv) [inline tests/llm/services/conftest.py:39]: Revert this change

**Review by mdemoret-nv — APPROVED** [review]

File: tests/llm/services/test_openai_chat_client.py (hunk 3 above)
Comment (mdemoret-nv) [reply tests/llm/services/test_openai_chat_client.py]: Will be cleaned up in #1704

**Pull request merged**